src/go/configgenerator/filter_generator.go

```diff
@@ -28,68 +28,50 @@ type FilterGenerator interface {
 	// FilterName returns the name of the filter.
 	FilterName() string
 
-	// GenFilterConfig generates the filter config and a list of methods that need per-route configs.
-	GenFilterConfig(*ci.ServiceInfo) (*hcmpb.HttpFilter, []*ci.MethodInfo, error)
+	// IsEnabled returns true if the filter config should be generated.
+	// If false, none of the generation methods will be called.
+	IsEnabled() bool
+
+	// GenFilterConfig generates the filter config.
+	//
+	// Return (nil, nil) if the filter has no listener-level config, but may
+	// have per-route configurations.
+	GenFilterConfig(*ci.ServiceInfo) (*hcmpb.HttpFilter, error)
 
 	// GenPerRouteConfig generates the per-route config for the given HTTP route.
-	// This method is only called on the routes that `GenFilterConfig` returns.
+	//
+	// This method is called on all routes. Return (nil, nil) to indicate the
```

Comment:
Is it called on the methodInfo? Is that equal to route?

Reply:
MethodInfo has multiple routes (HTTP patterns). This function generates config on the HTTP pattern, but the wrapping MethodInfo is also useful to have for generation.

Reply:
BTW this function does NOT generate route configs for all HTTP patterns under a method. It generates it for a single HTTP pattern at a time. Doing it one-at-a-time makes it easier for caller to fill in route table config.